optimum/exporters/onnx/__main__.py

```diff
@@ -333,6 +338,30 @@ def main_export(
                 f"The task could not be automatically inferred as this is available only for models hosted on the Hugging Face Hub. Please provide the argument --task with the relevant task from {', '.join(TasksManager.get_all_tasks())}. Detailed error: {e}"
             )
 
+    if library_name == "transformers":
+        config = AutoConfig.from_pretrained(
+            model_name_or_path,
+            subfolder=subfolder,
+            revision=revision,
+            cache_dir=cache_dir,
+            use_auth_token=use_auth_token,
+            local_files_only=local_files_only,
+            force_download=force_download,
+            trust_remote_code=trust_remote_code,
+        )
+        model_type = config.model_type.replace("_", "-")
+        if model_type not in TasksManager._SUPPORTED_MODEL_TYPE:
+            custom_architecture = True
```

Comment:
@fxmarty this line currently does nothing [since it is set to False again in line 381](https://github.com/huggingface/optimum/pull/1404/files#diff-d914aff54c87e54a36200bf46f382f111fb2bd34e66282e541e0be5f07e891eaR381). Do you want to have a look?

Reply:
Good catch! I'll fix